src/convolution.cpp

```diff
@@ -336,19 +423,12 @@ ConvolutionDescriptor::WrwGetValidWorkSpaceSizeGemm(const TensorDescriptor& dyDe
     return 0;
 }
 
-std::size_t ConvolutionDescriptor::ForwardGetWorkSpaceSize(Handle& handle,
-                                                           const TensorDescriptor& wDesc,
-                                                           const TensorDescriptor& xDesc,
-                                                           const TensorDescriptor& yDesc) const
+std::size_t ConvolutionDescriptor::GetWorkSpaceSize(const ExecutionContext& ctx_,
+                                                    const conv::ProblemDescription& problem) const
 {
     MIOPEN_LOG_I("");
 
-    auto ctx = ConvolutionContext{xDesc, wDesc, yDesc, *this, conv::Direction::Forward};
-    ctx.SetStream(&handle);
-    ctx.DetectRocm();
-    ctx.SetupFloats();
-    ctx.do_search             = false;
-    ctx.disable_perfdb_access = true;
+    auto ctx = ctx_.WithTuning(false).WithPerfdbAccess(false);
```

Comment:
@DrizztDoUrden Here is a double copy. Is that how it's intended?

Reply:
The first copy should get elided. But I should probably make ctx const here.

Reply:
My bad here, will fix.